File: 1750-minimum-length-of-string-after-deleting-similar-ends/1750-minimum-length-of-string-after-deleting-similar-ends.py
class Solution(object):
    def minimumLength(self, s):
        """
        :type s: str
        :rtype: int
        """
        sArr = list(s)
        left, right = 0, len(sArr) - 1
        while left < right:
            if sArr[left] != sArr[right]:
                return right - left + 1
            toBeTaken = sArr[left]
            while sArr[left] == toBeTaken and left < right:
                left += 1
            while sArr[right] == toBeTaken and left <= right:
                right -= 1
        return right - left + 1

File: 1750-minimum-length-of-string-after-deleting-similar-ends/1750-minimum-length-of-string-after-deleting-similar-ends.ts
/**
 * @param {string} s
 * @return {number}
 */
var minimumLength = function(s) {
    const sArr = s.split('');
    let left = 0, right = sArr.length - 1;
    let toBeTaken;
    
    while (left < right) {
        if (sArr[left] !== sArr[right]) return right - left + 1;
        
        toBeTaken = sArr[left];
        
        while (sArr[left] === toBeTaken && left < right) {
            left++;
        }
        
        while (sArr[right] === toBeTaken && left <= right) {
            right--;
        }
    }
    
    return right - left + 1;
};